pkg/controller/seed-controller-manager/backup/backup_controller.go

```diff
@@ -236,7 +236,7 @@ func (r *Reconciler) reconcile(ctx context.Context, log *zap.SugaredLogger, conf
 
 	// the etcdbackup/etcdrestore controllers are enabled (at least for this seed), so we
 	// only perform cleanup for older clusters when needed
-	controllerEnabled := !seed.IsDefaultEtcdAutomaticBackupEnabled()
+	controllerEnabled := seed.IsEtcdAutomaticBackupEnabled()
```

Comment:
This part was very misleading for me. Better name to the variable before the change would be `controllerDisabled` and I don't fully understand what this was trying to achieve.

Reply:
Now if the cluster is deleted and controller is running it schedules cleanup job.

Reply:
Wait, I think there's something wrong here. I believe this is an old backup controller and that is why it was checking for the "automatic backup" to be _disabled_, since it shouldn't run when the new one is enabled. I think this shouldn't change.

Reply:
Oh, now it makes more sense to me. Let me fix this.